packages/extension-ui/src/util/buildHierarchy.ts

```diff
@@ -3,18 +3,70 @@
 
 import { AccountJson, AccountWithChildren } from '@polkadot/extension-base/background/types';
 
+import getNetworkMap from './getNetworkMap';
+
 type ChildFilter = (account: AccountJson) => AccountWithChildren;
 
 function compareByCreationTime (a: AccountJson, b: AccountJson): number {
   return (a.whenCreated || Infinity) - (b.whenCreated || Infinity);
 }
 
+function compareByString (a?: string | null, b?: string | null): number {
+  const nameA = a?.toUpperCase() || '';
+  const nameB = b?.toUpperCase() || '';
+
+  if (nameA < nameB) {
+    return -1;
+  }
+
+  if (nameA > nameB) {
+    return 1;
+  }
+
+  // names are equal
+  return 0;
+}
+
+function compareByName (a: AccountJson, b: AccountJson): number {
+  return compareByString(a.name, b.name);
+}
+
+function compareByPath (a: AccountJson, b: AccountJson): number {
+  return compareByString(a.suri, b.suri);
+}
+
+function compareByNetwork (a: AccountJson, b: AccountJson): number {
+  const networkMap = getNetworkMap();
+  const networkA = networkMap.get(a?.genesisHash || '');
+  const networkB = networkMap.get(b?.genesisHash || '');
+
+  return compareByString(networkA, networkB);
+}
+
+function compareByNameThenCreation (a: AccountJson, b: AccountJson): number {
+  // This comparison happens after an initial sorting by network.
+  // if the 2 accounts are from different networks, don't touch their order
+  if (a.genesisHash !== b.genesisHash) {
+    return 0;
+  }
+
+  const res = compareByName(a, b);
+
+  return res === 0 ? compareByCreationTime(a, b) : res;
```

Comment:
Should move this into path since it gets done last.

Reply:
Yes, if we want to apply the same type of comparison Name > Path > Creation for each. I had Name > Creation for root and Name > Path for children (no Creation). Will change that then.